### PR TITLE
Fix partitioned table planning

### PR DESCRIPTION
**Pull request by amogh-jahagirdar**

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/UnboundBaseFileScanTask.java [hunk 1]
```diff
@@ -21,13 +21,15 @@
 import org.apache.iceberg.expressions.Expression;
 import org.apache.iceberg.expressions.ResidualEvaluator;
 
-public class UnboundBaseFileScanTask extends BaseFileScanTask {
```
Comment (amogh-jahagirdar) [inline core/src/main/java/org/apache/iceberg/UnboundBaseFileScanTask.java:24]: this is important, we should double check we're not making classes unnecessarily public. I cleaned up this case, but let's double check as we iterate.

File: core/src/main/java/org/apache/iceberg/rest/responses/FetchScanTasksResponseParser.java [hunk 2]
```diff
@@ -43,7 +43,9 @@ public static String toJson(FetchScanTasksResponse response, boolean pretty) {
   public static void toJson(FetchScanTasksResponse response, JsonGenerator gen) throws IOException {
     Preconditions.checkArgument(response != null, "Invalid response: fetchScanTasksResponse null");
     Preconditions.checkArgument(
-        response.specsById() != null, "Cannot serialize fetchScanTasksResponse without specsById");
+        response.specsById() != null
+            || (response.fileScanTasks() == null || response.fileScanTasks().isEmpty()),
```
Comment (amogh-jahagirdar) [inline core/src/main/java/org/apache/iceberg/rest/responses/FetchScanTasksResponseParser.java:47]: I fixed the condition for failing when specs by id is missing as well. We only need to have specs by Id defined when the response has file scan tasks, but it's not needed if there's just plan tasks.

File: core/src/main/java/org/apache/iceberg/UnboundBaseFileScanTask.java (hunk 1 above)
Comment (rahil-c) [reply core/src/main/java/org/apache/iceberg/UnboundBaseFileScanTask.java:24]: ack

File: core/src/main/java/org/apache/iceberg/rest/responses/FetchScanTasksResponseParser.java (hunk 2 above)
Comment (rahil-c) [reply core/src/main/java/org/apache/iceberg/rest/responses/FetchScanTasksResponseParser.java:47]: Nice catch thanks!

**Review by rahil-c — APPROVED** [review]
Thanks @amogh-jahagirdar this looks clean!

**Pull request merged**